internals/models.py

```diff
@@ -954,18 +954,68 @@ def get_in_milestone(
     q = q.order(Feature.name)
     q = q.filter(Feature.shipped_milestone == milestone)
     desktop_shipping_features = q.fetch(None)
+    for feature in desktop_shipping_features:
+      if feature.impl_status_chrome not in {ENABLED_BY_DEFAULT, DEPRECATED, REMOVED}:
+        if feature.feature_type == FEATURE_TYPE_DEPRECATION_ID and Feature.dt_milestone_desktop_start != None:
+          feature.impl_status_chrome = DEPRECATED
```

Comment:
FYI, as for why the old features.json response was structure as a flat list with annotations: it might have been evolved from the idea of serving an RSS feed, which is limited to being a flat list.   I don't know, that was before my time on the project.

Reply:
Okay. I think I get it. It does seem to be a better way. I will make this change.

Reply:
However, I would like to add that I am setting `impl_status_chrome` in each feature because the frontend maps each feature to shipping type to create a JS object similar to JSON response suggested by you. In fact, we do not even have to annotate the list and hence do not have to sort the list by name or implementation status at the backend.

Reply:
Even the old releases page maps each feature to shipping type at the frontend although it receives the annotated flat list as it uses the `get_all_features()` API (also used by feature list page), instead of the new `get_features_in_milestone()` API.

Reply:
So, all this means that the change suggested by you will result in moving the process of mapping features to shipping type from frontend to backend.

Reply:
It still makes sense to make this change. However, it would be better to implement it in a separate PR. It is so because implementing this change involves making changes in the same files that #1458 changes. Since #1458 introduces structural changes, making changes in this PR would be like making changes without seeing the source code.

Reply:
OK, that sounds fine.